### PR TITLE
API-9011: [Form526 EVSS Validations] - "veteran.changeOfAddress"

### PR DESCRIPTION
**Pull request by SlakrHakr**
## Description of change
Adds some validation around the changeOfAddress portion of a 526 submission, based on information given to us from EVSS.

## Original issue(s)
https://vajira.max.gov/browse/API-9011

## Things to know about this PR
Requires this PR for sandbox (and any other mocked environment)
https://github.com/department-of-veterans-affairs/vets-api-mockdata/pull/174

### CODE REVIEW

File: modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb [hunk 1]
```diff
@@ -143,6 +143,140 @@
...
+              addressLine1: '1234 Couch Street',
+              city: 'New York City',
+              state: 'NY',
+              type: 'DOMESTIC',
+              zipFirstFive: '12345',
+              country: 'BlahBlahBlah'
+            }
+          end
+
+          it 'raises an exception that endingDate is provided but should not be' do
```
Comment (jw81) [inline modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb:262]: Probably want this spec name to be something related to the `country` attribute being invalid.

File: modules/claims_api/config/schemas/526.json [hunk 2]
```diff
@@ -171,12 +171,18 @@
...
             "beginningDate": {
               "description": "Date in YYYY-MM-DD the Veteran changed address.",
               "type": "string",
               "example": "2018-06-04"
             },
+            "endingDate": {
+              "description": "Date in YYYY-MM-DD the changed address expires, if change is temporary.",
+              "type": "string",
+              "example": "2018-06-04"
+            },
```
Comment (jw81) [inline modules/claims_api/config/schemas/526.json:185]: Should we apply a date pattern to this, similar to [here](https://github.com/department-of-veterans-affairs/vets-api/blob/2fe6a2ce2355cd5d05d2ec5d04c8698c76f7f9dc/modules/claims_api/config/schemas/526.json#L464)?

Comment (SlakrHakr) [reply modules/claims_api/config/schemas/526.json:187]: done. also removed all the XXXX-XX-XX from other patterns
https://github.com/department-of-veterans-affairs/vets-api/pull/7711/commits/0fe8d82fb309953deeb237eb0bc08e1e986c157f

File: modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb (hunk 1 above)
Comment (SlakrHakr) [reply modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb]: done
https://github.com/department-of-veterans-affairs/vets-api/pull/7711/commits/68c9fe826a7eb327303ddcf51e4a12541d3d6ad3

**Review by jw81 — APPROVED** [review]

**Pull request merged**